lib/board.rb

```diff
@@ -38,7 +33,15 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
-    ship.length == coordinates.length && valid_coordinate_array?(coordinates) == true
+    ship.length == coordinates.length && valid_coordinate_array?(coordinates) == true && cells_not_empty?(coordinates) == false
+  end
+
+
+  def cells_not_empty?(coordinates)
+    overlap = coordinates.all? do |coordinate|
+      coordinate.empty?
+    end
+    overlap
```

Comment:
I think I need to write more tests to confirm this works they way I intend. The boolean is always the opposite of what I think it should be and I don't know if that is  part of .all? works or I've done something horribly wrong. For now, though, all the tests are passing.